app/main/controllers/activity/RTMediaActivity.php

```diff
@@ -34,16 +34,21 @@ function __construct( $media, $privacy = 0, $activity_text = false ) {
 		$this->privacy       = $privacy;
 	}
 
+	/**
+	 * Create html of activity/comments using text and media.
+	 * 
+	 * @param string $type Type of activity (activity/comment).
+	 */
 	function create_activity_html( $type = 'activity' ) {
 
-		$html = '';
-
-		$html .= '<div class="rtmedia-'.$type.'-container">';
+		$activity_container_start = '<div class="rtmedia-'. esc_attr( $type ) .'-container">';
+		$activity_container_end   = '</div>';
 
+		$activity_text = '';
 		if ( ! empty( $this->activity_text ) && '&nbsp;' !== $this->activity_text ) {
-			$html .= '<div class="rtmedia-'.$type.'-text"><span>';
-			$html .= $this->activity_text;
-			$html .= '</span></div>';
+			$activity_text .= '<div class="rtmedia-' . esc_attr( $type ) . '-text"><span>';
```

Comment:
As we are redefining this whole function can we use sprintf in all the html for more readability?

`$activity_text .= sprintf( '<div class="rtmedia-%s-text"><span>', esc_attr( $type ) );`

Reply:
we can use this in one line. 

$activity_text .= sprintf( '<div class="rtmedia-%1$s-text"><span>%2$s</span></div>', esc_attr( $type ),  $this->activity_text );

check if require escaping in activity text.